test/arena_test.dart

```diff
@@ -163,6 +163,27 @@ void main() async {
     expect(freed.single, 1234);
   });
 
+  test('zone future error', () async {
+    bool caughtError = false;
+    bool uncaughtError = false;
+    final future = runZonedGuarded(() {
+      return withZoneArena(() async {
+        throw Exception('Exception 4');
+        // ignore: dead_code
```

Comment:
Why do you have dead code here?
If the inference gets in your way, could you just provide a type argument to `withZoneArena`? (Would it be `withZoneArena<Future<int>>`?)